### ISSUE TITLE
md-cell: 'required' causes error when md-table is inside of ui-view, which is used by child state

### ISSUE THREAD

**Issue by anatolyz** [daniel-nagy/md-data-table#325]
Hi,
I found some issue, similar to https://github.com/angular/material/issues/2414 bug on angular-material.

I have nested states and some **parent** and **child** state. **Parent** has ui-view (let's name in *parent-state-view*) with md-table inside. Everything works fine.
But. I have **child** state, which uses *parent-state-view*. And when I open this **child** state, md-cell throws error.
See http://plnkr.co/edit/tofIeDIiC88P4H5h4itP?p=preview

Can we resolve this problem like md bug above? https://github.com/angular/material/commit/4350803267a8e19a15a240ccf657cd2ba098e0ae

**Comment by daniel-nagy** [comment]
@anatolyz I deferred your plunker to that issue you referenced. I believe they should fix this on their end but if they don't I'll do something similar to what the Angular Material team did.

**Comment by anatolyz** [comment]
What is the cause of this error? Angular-material?

**Comment by daniel-nagy** [comment]
No, I think they way ui-view replaces the content causes the error.

**Comment by mikedon** [comment]
Has there been any progress on this?  I need this for a project at work.

I've tested out the changes locally to not require the tableCtrl and put some if checks in place to avoid undefined access.  Not sure if I'm covering all the cases but it seems to get past the error for me.  Want a PR?

**Comment by iKarl** [comment]
I have the same problem as @anatolyzolotar, any help?

**Comment by jbh** [comment]
+1!!!

**Comment by LeonVay** [comment]
Hello everyone,
I have faced the same problem: Controller 'mdDataTable', required by directive 'mdTableRow', can't be found!
On different versions of angular (1.5.X) there are different messages, but the root cause seems to be the same.
Switching off animation does not helped.

**Comment by elekzalan** [comment]
@daniel-nagy Do you have any progress on this issue? What do you think about @mikedon 's workaround?

**Comment by mattbajorek** [comment]
Check the name of your component/child.  Make sure its not a reserved html tag.
